MLS.Agent/CommandLine/VerifyCommand.cs

```diff
@@ -23,11 +23,11 @@ public static class VerifyCommand
         public static async Task<int> Do(
             VerifyOptions options,
             IConsole console,
-            Func<IDirectoryAccessor> getDirectoryAccessor,
-            PackageRegistry packageRegistry,
+            Func<DirectoryInfo, IDirectoryAccessor> getDirectoryAccessor,
             StartupOptions startupOptions = null)
         {
-            var directoryAccessor = getDirectoryAccessor();
+            var packageRegistry = PackageRegistry.CreateForTryMode(options.Dir);
+            var directoryAccessor = getDirectoryAccessor(options.Dir);
```

Comment:
This pattern is also a little weird--could we push the directoryAccessor into the options?

Reply:
The options are used at the command line and they take the directory from the base `dotnet try` command. I was unable to determine how to convert the dirArgument from the base command to a DirectoryAccessor for the subcommand. There is a method called TryConvertArgument, but I couldnt figure out how to use it to get the desired result. Maybe @jonsequitur might have more ideas.

For now, I have used the dir from the VerifyOptions and then created a DirectoryAccessor which is passed on to the VerifyCommand, hence VerifyCommand now takes only a directory accessor and no directory.

Reply:
Ideally `PackageRegistry.CreateForTryMode` would also take in `IDirectoryAccessor` and not `DirectoryInfo`.